### PR TITLE
Try to fix Windows CI

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Follow-up of #985.

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -124,10 +124,26 @@ jobs:
         run: sudo apt install libgtk-3-dev libssh2-1-dev libglib2.0-dev libgraphene-1.0-dev libcairo-gobject2 libcairo2-dev
         if: matrix.os == 'ubuntu-20.04'
       - name: Install toolchain packages with pacman
-        run: pacman --noconfirm -S base-devel mingw-w64-${{ matrix.arch }}-toolchain
+        steps:
+          - id: step1
+            run curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
```
Comment (bilelmoussaoui) [inline .github/workflows/main.yaml:129]:
```suggestion
            run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
```

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -124,10 +124,16 @@ jobs:
         run: sudo apt install libgtk-3-dev libssh2-1-dev libglib2.0-dev libgraphene-1.0-dev libcairo-gobject2 libcairo2-dev
         if: matrix.os == 'ubuntu-20.04'
       - name: Install toolchain packages with pacman
-        run: pacman --noconfirm -S base-devel mingw-w64-${{ matrix.arch }}-toolchain
+        run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
+        run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz.sig
+        run: pacman -U --noconfirm --config <(echo) msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
+        run: pacman --noconfirm -Syu base-devel mingw-w64-${{ matrix.arch }}-toolchain
```
Comment (bilelmoussaoui) [inline .github/workflows/main.yaml:130]:
```suggestion
          run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
        - run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz.sig
        - run: pacman -U --noconfirm --config <(echo) msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
        - run: pacman --noconfirm -Syu base-devel mingw-w64-${{ matrix.arch }}-toolchain
```
The first run block has a name

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -124,10 +124,16 @@ jobs:
         run: sudo apt install libgtk-3-dev libssh2-1-dev libglib2.0-dev libgraphene-1.0-dev libcairo-gobject2 libcairo2-dev
         if: matrix.os == 'ubuntu-20.04'
       - name: Install toolchain packages with pacman
-        run: pacman --noconfirm -S base-devel mingw-w64-${{ matrix.arch }}-toolchain
+        run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
+        - run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz.sig
+        - run: pacman -U --noconfirm --config <(echo) msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
+        - run: pacman --noconfirm -Syu base-devel mingw-w64-${{ matrix.arch }}-toolchain
```
Comment (bilelmoussaoui) [inline .github/workflows/main.yaml:130]:
```suggestion
        - run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz.sig
        - run: pacman -U --noconfirm --config <(echo) msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
        - run: pacman --noconfirm -Syu base-devel mingw-w64-${{ matrix.arch }}-toolchain
       
```
those are mis-indentended as well

**Comment by bilelmoussaoui** [comment]
I think something like this should work
```
      - name: Install packages from apt
        run: sudo apt install libgtk-3-dev libssh2-1-dev libglib2.0-dev libgraphene-1.0-dev libcairo-gobject2 libcairo2-dev
        if: matrix.os == 'ubuntu-20.04'
      - name: Install toolchain packages with pacman
        run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
        if: matrix.os == 'windows-latest'
      - run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz.sig
        if: matrix.os == 'windows-latest'
      - run: pacman -U --noconfirm --config <(echo) msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
        if: matrix.os == 'windows-latest'
      - run: pacman --noconfirm -Syu base-devel mingw-w64-${{ matrix.arch }}-toolchain
        if: matrix.os == 'windows-latest'
      - name: Install library devel packages with pacman
        run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
        if: matrix.os == 'windows-latest'
      - run: curl -O http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz.sig
        if: matrix.os == 'windows-latest'
      - run: pacman -U --noconfirm --config <(echo) msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz
        if: matrix.os == 'windows-latest'
      - run: pacman --noconfirm -Syu mingw-w64-${{ matrix.arch }}-gtk3 mingw-w64-${{ matrix.arch }}-libgit2 mingw-w64-${{ matrix.arch }}-libsecret
        if: matrix.os == 'windows-latest'
```
I didn't notice the if matrix.os=='window-latest' before... I think what would make sense is to split the job into a windows/linux one instead of having a bunch of conditions though

**Comment by sdroege** [comment]
Should we just disable the Windows CI for now here? gir doesn't do anything Windows-specific anyway, it would seem more useful to have a Windows CI for gtk-rs instead.

**Comment by GuillaumeGomez** [comment]
I will comment out all the windows part. It's just broken beyond recovery and I think it'll be for a while.

**Pull request merged**